distributed/tests/test_s3.py

```diff
@@ -393,3 +394,91 @@ def test_read_csv_sync(loop):
             assert list(df.columns) == ['name', 'amount', 'id']
             f = e.compute(df.amount.sum())
             assert f.result() == (100 + 200 + 300 + 400 + 500 + 600)
+
+def test_avro_body(s3):
+    fastavro = pytest.importorskip('fastavro')
+    fn = 'distributed-test/test/data/avro/2.avro'
+    byte = s3.cat(fn)
+    av = fastavro.reader(s3.open(fn))
+    header = av._header
+    records = list(av)
+    
+    schema = header['meta']['avro.schema'].decode()
+    schema = json.loads(schema)
+    assert ['amount', 'name'] == sorted(f['name'] for f in schema['fields'])
+    sync = header['sync']
+    assert len(sync) == 16
+    bit = byte[byte.index(sync)+len(sync):]
+    assert avro_body(bit, header) == records
+    bit = byte[byte.index(sync)+len(sync):-len(sync)]
+    assert avro_body(bit, header) == records
+    assert avro_body(b"", header) == []
+
+
+def test_avro_to_df(s3):
+    fastavro = pytest.importorskip('fastavro')
+    pytest.importorskip('pandas')
+    fn = 'distributed-test/test/data/avro/2.avro'
+    byte = s3.cat(fn)
+    av = fastavro.reader(s3.open(fn))
+    header = av._header
+    sync = header['sync']
+    bit = byte[byte.index(sync)+len(sync):]
+
+    df = avro_to_df(bit, header)
+    assert ['amount', 'name'] == sorted(df.columns)
+    assert len(df) == 4
+    assert 100 + 200 + 300 + 400 == df.amount.sum()
+    df = avro_to_df(b"", header)
+    assert len(df) == 0
+    df = avro_to_df(b"", header, cols=['amount', 'name'])
+    assert ['amount', 'name'] == list(df.columns)
```

Comment:
Same here, this function takes bytes.  Ideally a unittest relies only on the functionality we intend to test.  At the moment this test, which I think intends to test `avro_to_df` will break if any of the S3 code breaks as well.